Use more smart pointers instead of raw pointers

achristensen07 opened the pull request:
#### cd352c23e71a20eb51389b4b3845293c78432a1c
<pre>
Use more smart pointers instead of raw pointers
<a href="https://bugs.webkit.org/show_bug.cgi?id=258046">https://bugs.webkit.org/show_bug.cgi?id=258046</a>
rdar://110734344

Reviewed by David Kilzer.

* Source/WebCore/Modules/airplay/WebMediaSessionManager.cpp:
(WebCore::WebMediaSessionManager::showPlaybackTargetPicker):
* Source/WebCore/Modules/airplay/WebMediaSessionManagerClient.h:
* Source/WebCore/Scripts/SettingsTemplates/InternalSettingsGenerated.h.erb:
* Source/WebCore/inspector/InspectorFrontendClientLocal.cpp:
(WebCore::InspectorFrontendClientLocal::frontendPage):
(WebCore::InspectorFrontendClientLocal::windowObjectCleared):
* Source/WebCore/inspector/InspectorFrontendClientLocal.h:
(WebCore::InspectorFrontendClientLocal::frontendPage const): Deleted.
* Source/WebCore/inspector/InspectorFrontendHost.h:
* Source/WebCore/page/PageOverlay.cpp:
(WebCore::PageOverlay::page const):
* Source/WebCore/page/PageOverlay.h:
* Source/WebCore/page/SettingsBase.cpp:
(WebCore::SettingsBase::setStandardFontFamily):
(WebCore::SettingsBase::setFixedFontFamily):
(WebCore::SettingsBase::setSerifFontFamily):
(WebCore::SettingsBase::setSansSerifFontFamily):
(WebCore::SettingsBase::setCursiveFontFamily):
(WebCore::SettingsBase::setFantasyFontFamily):
(WebCore::SettingsBase::setPictographFontFamily):
* Source/WebCore/page/SettingsBase.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
* Source/WebCore/testing/InternalSettings.h:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
(WebKit::WebPageProxy::Internals::platformView const):
* Source/WebKit/WebProcess/InjectedBundle/API/c/mac/WKBundlePageBannerMac.mm:
(WKBundlePageBannerCreateBannerWithCALayer):
* Source/WebKit/WebProcess/Inspector/WebInspector.cpp:
(WebKit::WebInspector::page const):
* Source/WebKit/WebProcess/Inspector/WebInspector.h:
(WebKit::WebInspector::page const): Deleted.
* Source/WebKit/WebProcess/Inspector/WebInspectorClient.h:
* Source/WebKit/WebProcess/Notifications/NotificationPermissionRequestManager.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebAlternativeTextClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebColorChooser.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebContextMenuClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebDragClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebEditorClient.cpp:
(WebKit::WebEditorClient::textFieldDidBeginEditing):
(WebKit::WebEditorClient::textFieldDidEndEditing):
(WebKit::WebEditorClient::textDidChangeInTextField):
(WebKit::WebEditorClient::textDidChangeInTextArea):
(WebKit::WebEditorClient::doTextFieldCommandFromEvent):
(WebKit::WebEditorClient::textWillBeDeletedInTextField):
* Source/WebKit/WebProcess/WebCoreSupport/WebEditorClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebNotificationClient.cpp:
(WebKit::WebNotificationClient::show):
(WebKit::WebNotificationClient::cancel):
(WebKit::WebNotificationClient::notificationObjectDestroyed):
* Source/WebKit/WebProcess/WebCoreSupport/WebNotificationClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebPopupMenu.h:
(WebKit::WebPopupMenu::page):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebEditorClientMac.mm:
(WebKit::WebEditorClient::uppercaseWord):
(WebKit::WebEditorClient::lowercaseWord):
(WebKit::WebEditorClient::capitalizeWord):
* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::findString):
(WebKit::FindController::findStringMatches):
(WebKit::FindController::findRectsForStringMatches):
* Source/WebKit/WebProcess/WebPage/FindController.h:
* Source/WebKit/WebProcess/WebPage/PageBanner.cpp:
(WebKit::PageBanner::~PageBanner):
* Source/WebKit/WebProcess/WebPage/PageBanner.h:
(WebKit::PageBanner::Client::~Client):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteScrollingCoordinator.h:
* Source/WebKit/WebProcess/WebPage/WebBackForwardListProxy.h:
* Source/WebKit/WebProcess/WebPage/WebOpenPanelResultListener.h:
* Source/WebKit/WebProcess/WebPage/mac/PageBannerMac.mm:
(WebKit::PageBanner::create):
(WebKit::PageBanner::PageBanner):
(WebKit::PageBanner::showIfHidden):
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationScrollingCoordinator.h:
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.h:
* Source/WebKit/WebProcess/cocoa/VideoFullscreenManager.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.mm:
(WebInspectorClient::openLocalFrontend):
(WebInspectorClient::didSetSearchingForNode):
* Source/WebKitLegacy/mac/WebView/WebMediaPlaybackTargetPicker.h:
* Source/WebKitLegacy/mac/WebView/WebMediaPlaybackTargetPicker.mm:
(WebMediaPlaybackTargetPicker::platformView const):

Canonical link: <a href="https://commits.webkit.org/265162@main">https://commits.webkit.org/265162@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ade41d655b1970436bc98ec3bd125770c5258a24

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10063 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10307 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10560 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11714 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/9731 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10072 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/12296 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10259 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12673 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10218 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10991 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8518 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12099 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8297 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9123 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16445 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9399 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9274 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12533 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9763 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/7929 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8922 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13155 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1132 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9550 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer commented on Source/WebCore/inspector/InspectorFrontendClientLocal.cpp:
```diff
@@ -171,12 +171,17 @@ void InspectorFrontendClientLocal::resetState()
     m_settings->deleteProperty(inspectorAttachedHeightSetting);
 }
 
+Page* InspectorFrontendClientLocal::frontendPage()
+{
+    return m_frontendPage.get();
+}
+
```
Can this be `const`?

If this would require changing the whole class hierarchy, does not have to be done in this patch.

achristensen07 replied:
It would require changing lots of things.  Won't do here.

ddkilzer commented on Source/WebCore/inspector/InspectorFrontendClientLocal.cpp:
```diff
@@ -171,12 +171,17 @@ void InspectorFrontendClientLocal::resetState()
         m_frontendHost->disconnectClient();
     
-    m_frontendHost = InspectorFrontendHost::create(this, m_frontendPage);
+    m_frontendHost = InspectorFrontendHost::create(this, m_frontendPage.get());
```
Use `frontendPage()` here?

ddkilzer commented on Source/WebCore/page/SettingsBase.cpp:
```diff
@@ -109,7 +109,7 @@ void SettingsBase::setStandardFontFamily(const String& family, UScriptCode scrip
     bool changes = fontGenericFamilies().setStandardFontFamily(family, script);
     if (changes)
-        invalidateAfterGenericFamilyChange(m_page);
+        invalidateAfterGenericFamilyChange(m_page.get());
```
Use page() in `SettingsBase.cpp`?

achristensen07 replied:
There is no page() in SettingsBase.

ddkilzer commented on Source/WebKit/WebProcess/WebPage/FindController.cpp:
```diff
@@ -300,7 +300,7 @@ void FindController::findString(const String& string, OptionSet<FindOptions> opt
 #endif
 
-    RefPtr<WebPage> protectedWebPage = m_webPage;
+    RefPtr<WebPage> protectedWebPage { m_webPage.get() };
```
Can we create a convenience `RefPtr::RefPtr<T>(WeakPtr<T>)` constructor so we don't even have to use `get()` here?

Not required to land the patch.

achristensen07 replied:
That might be nice.  Not all WeakPtr types can make a RefPtr, but we could probably do it.

ddkilzer commented on Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.mm:
```diff
@@ -161,7 +161,7 @@ - (void)destroyInspectorView;
     if (enabled) {
         [[m_inspectedWebView window] makeKeyAndOrderFront:nil];
-        [[m_inspectedWebView window] makeFirstResponder:m_inspectedWebView];
+        [[m_inspectedWebView window] makeFirstResponder:m_inspectedWebView.get().get()];
```
Using `[m_inspectedWebView window]` with `m_inspectedWebView` as a `WeakObjCPtr<>` causes each method call to autorelease the WebView.

One way to fix this would be to add a convenience method like this, then use that everywhere:

```
    RetainPtr<WebView> inspectedWebView() const { return m_inspectedWebView.get(); }
```

Not necessary to make this change, but I wanted you to be aware of the behavior change.

achristensen07 replied:
I added a few .get()'s to not add to the autorelease pool.

ddkilzer reviewed (approved):
r=me.  None of the feedback is required to land, but are suggestions for minor improvements.

achristensen07 committed:
Use more smart pointers instead of raw pointers
https://bugs.webkit.org/show_bug.cgi?id=258046
rdar://110734344

Reviewed by David Kilzer.

* Source/WebCore/Modules/airplay/WebMediaSessionManager.cpp:
(WebCore::WebMediaSessionManager::showPlaybackTargetPicker):
* Source/WebCore/Modules/airplay/WebMediaSessionManagerClient.h:
* Source/WebCore/Scripts/SettingsTemplates/InternalSettingsGenerated.h.erb:
* Source/WebCore/inspector/InspectorFrontendClientLocal.cpp:
(WebCore::InspectorFrontendClientLocal::frontendPage):
(WebCore::InspectorFrontendClientLocal::windowObjectCleared):
* Source/WebCore/inspector/InspectorFrontendClientLocal.h:
(WebCore::InspectorFrontendClientLocal::frontendPage const): Deleted.
* Source/WebCore/inspector/InspectorFrontendHost.h:
* Source/WebCore/page/PageOverlay.cpp:
(WebCore::PageOverlay::page const):
* Source/WebCore/page/PageOverlay.h:
* Source/WebCore/page/SettingsBase.cpp:
(WebCore::SettingsBase::setStandardFontFamily):
(WebCore::SettingsBase::setFixedFontFamily):
(WebCore::SettingsBase::setSerifFontFamily):
(WebCore::SettingsBase::setSansSerifFontFamily):
(WebCore::SettingsBase::setCursiveFontFamily):
(WebCore::SettingsBase::setFantasyFontFamily):
(WebCore::SettingsBase::setPictographFontFamily):
* Source/WebCore/page/SettingsBase.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
* Source/WebCore/testing/InternalSettings.h:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
(WebKit::WebPageProxy::Internals::platformView const):
* Source/WebKit/WebProcess/InjectedBundle/API/c/mac/WKBundlePageBannerMac.mm:
(WKBundlePageBannerCreateBannerWithCALayer):
* Source/WebKit/WebProcess/Inspector/WebInspector.cpp:
(WebKit::WebInspector::page const):
* Source/WebKit/WebProcess/Inspector/WebInspector.h:
(WebKit::WebInspector::page const): Deleted.
* Source/WebKit/WebProcess/Inspector/WebInspectorClient.h:
* Source/WebKit/WebProcess/Notifications/NotificationPermissionRequestManager.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebAlternativeTextClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebColorChooser.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebContextMenuClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebDragClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebEditorClient.cpp:
(WebKit::WebEditorClient::textFieldDidBeginEditing):
(WebKit::WebEditorClient::textFieldDidEndEditing):
(WebKit::WebEditorClient::textDidChangeInTextField):
(WebKit::WebEditorClient::textDidChangeInTextArea):
(WebKit::WebEditorClient::doTextFieldCommandFromEvent):
(WebKit::WebEditorClient::textWillBeDeletedInTextField):
* Source/WebKit/WebProcess/WebCoreSupport/WebEditorClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebNotificationClient.cpp:
(WebKit::WebNotificationClient::show):
(WebKit::WebNotificationClient::cancel):
(WebKit::WebNotificationClient::notificationObjectDestroyed):
* Source/WebKit/WebProcess/WebCoreSupport/WebNotificationClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebPopupMenu.h:
(WebKit::WebPopupMenu::page):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebEditorClientMac.mm:
(WebKit::WebEditorClient::uppercaseWord):
(WebKit::WebEditorClient::lowercaseWord):
(WebKit::WebEditorClient::capitalizeWord):
* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::findString):
(WebKit::FindController::findStringMatches):
(WebKit::FindController::findRectsForStringMatches):
* Source/WebKit/WebProcess/WebPage/FindController.h:
* Source/WebKit/WebProcess/WebPage/PageBanner.cpp:
(WebKit::PageBanner::~PageBanner):
* Source/WebKit/WebProcess/WebPage/PageBanner.h:
(WebKit::PageBanner::Client::~Client):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteScrollingCoordinator.h:
* Source/WebKit/WebProcess/WebPage/WebBackForwardListProxy.h:
* Source/WebKit/WebProcess/WebPage/WebOpenPanelResultListener.h:
* Source/WebKit/WebProcess/WebPage/mac/PageBannerMac.mm:
(WebKit::PageBanner::create):
(WebKit::PageBanner::PageBanner):
(WebKit::PageBanner::showIfHidden):
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationScrollingCoordinator.h:
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.h:
* Source/WebKit/WebProcess/cocoa/VideoFullscreenManager.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.mm:
(WebInspectorClient::openLocalFrontend):
(WebInspectorClient::didSetSearchingForNode):
* Source/WebKitLegacy/mac/WebView/WebMediaPlaybackTargetPicker.h:
* Source/WebKitLegacy/mac/WebView/WebMediaPlaybackTargetPicker.mm:
(WebMediaPlaybackTargetPicker::platformView const):

Canonical link: https://commits.webkit.org/265162@main
